src/test/java/com/proofpoint/discovery/InMemoryDynamicStore.java

```diff
@@ -99,39 +101,59 @@ public synchronized void delete(Id<Node> nodeId)
                 .filter(matchesType(type).and(matchesPool(pool)));
     }
 
+    @Override
+    public Entry get(Id<Node> nodeId)
+    {
+        removeExpired();
+
+        LocalEntry localEntry = descriptors.get(nodeId);
+        if (localEntry == null) {
+            return null;
+        }
+        long timestamp = localEntry.getExpiration().minusMillis((int) maxAge.toMillis()).toEpochMilli();
+        return Entry.entry(nodeId.getBytes(), new ArrayList<>(localEntry.getServices()), timestamp, maxAge.toMillis(), localEntry.getAnnouncer());
```

Comment:
localEntry.getServices() is immutable, so what's the need to make a mutable copy?